flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/AmbiguousTableFactoryException.java

```diff
@@ -31,7 +31,7 @@
 public class AmbiguousTableFactoryException extends RuntimeException {
 
 	// factories that match the properties
-	private final List<TableFactory> matchingFactories;
+	private final List<?> matchingFactories;
```

Comment:
Why do we need a wildcard here? All objects in this list are still table factories, no?

Reply:
This makes it easier to use with the subtypes of `TableFactory`, e.g. you can pass `List<PlannerFactory>`. I changed it though to `List<? extends TableFactory>`